ddp-workspace/projects/ddp-dsm-ui/src/app/tissue/tissue.component.ts

```diff
@@ -214,4 +284,189 @@ export class TissueComponent {
       this.currentPatchField = field;
     }
   }
+
+  changeSmId( event: any, parameterName, id, type, smIdArray, index, filedName? ) {
+    let value: any;
+    if (typeof event === 'string') {
+      value = event;
+    }
+    else if (event.srcElement != null && typeof event.srcElement.value === 'string') {
+      value = event.srcElement.value;
+    }
+    else if (typeof event === 'boolean') {
+      value = event;
+    }
+    else if (event.value != null) {
+      value = event.value;
+    }
+    if (filedName) {
+      this.currentPatchField = filedName;
+    }
+    if (!id) {
+      let smIdPk = this.valueChanged( type, 'smIdType', 'tissueId', this.tissue.tissueId, Statics.SM_ID_ALIAS, id, smIdArray, index, value, parameterName );
```

Comment:
should be changed to 
`if (!id) {
       this.valueChanged( type, "smIdType", "tissueId", this.tissue.tissueId, Statics.SM_ID_ALIAS, id, smIdArray, index, value, parameterName );

     }`

Reply:
can you please add the changes. thank you